# Cleansing Data Youtube Rewind 2015

#### Import Library

In [1]:
from dotenv import load_dotenv
from pymongo import MongoClient
import os
import pandas as pd

#### Import url dari .env

In [2]:
# Memuat value dari file .env
load_dotenv()

mongodb_url = os.getenv('URL_SANDY')
local_url = os.getenv('URL_LOCAL')

In [3]:
database_name = "youtube_rewind_indonesia"
collection_from = "collect_2015"
collection_name = "cleansing_2015"

#### Import dataset dari MongoDB

In [4]:
# Koneksi ke MongoDB
client = MongoClient(local_url)

# Pilih database dan koleksi
db = client[database_name]
collection = db[collection_from]

# Query data dari koleksi
cursor = collection.find()

# Konversi cursor ke list dan kemudian ke DataFrame
data = list(cursor)
df = pd.DataFrame(data)

# Menutup koneksi
client.close()

In [5]:
df

_id                 channelId      videoId  \
0      66515de230ea1caf68a1005c  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
1      66515de230ea1caf68a1005d  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
2      66515de230ea1caf68a1005e  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
3      66515de230ea1caf68a1005f  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
4      66515de230ea1caf68a10060  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
...                         ...                       ...          ...   
28409  66515e2230ea1caf68a16f55  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28410  66515e2230ea1caf68a16f56  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28411  66515e2230ea1caf68a16f57  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28412  66515e2230ea1caf68a16f58  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28413  66515e2230ea1caf68a16f59  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   

                                             textDisplay  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2      <a href="https://www.youtube.com/watch?v=fNX6G...   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28409                                          Mantep :v   
28410                                             Boom !   
28411                            anjing canggih waaahhhh   
28412                                              firat   
28413                                                  1   

                                            textOriginal  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2                                  3:34 scene yg gw suka   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28409                                          Mantep :v   
28410                                             Boom !   
28411                            anjing canggih waaahhhh   
28412                                              firat   
28413                                                  1   

            authorDisplayName  \
0         @KaylaAzzahra-fl9sf   
1             @rezapratama391   
2                @acesabo9179   
3                  @WINHERO10   
4            @nadyaamalia1486   
...                       ...   
28409       @GantplayBlogspot   
28410       @ramandahafiz6293   
28411                @e4rtz84   
28412  @pricillaoktaviani2376   
28413       @abayartsgame4609   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/ytc/AIdro_n7417ek-18wzJI...   
1      https://yt3.ggpht.com/ytc/AIdro_kt7csXE65KUFFl...   
2      https://yt3.ggpht.com/ytc/AIdro_lVQpw_BrtYzZYl...   
3      https://yt3.ggpht.com/2S9TD1tTD65kDm2AAWagbPeA...   
4      https://yt3.ggpht.com/ytc/AIdro_koGD1zvhTLlTLo...   
...                                                  ...   
28409  https://yt3.ggpht.com/ytc/AIdro_nyQqIuJWFK6Ie5...   
28410  https://yt3.ggpht.com/ytc/AIdro_nikMBBogL10baS...   
28411  https://yt3.ggpht.com/HWmDyGnG2Kzsde0_Z8ovBbWv...   
28412  https://yt3.ggpht.com/ytc/AIdro_lIX3-HU6zaCl0Z...   
28413  https://yt3.ggpht.com/ytc/AIdro_lwfz4EPeSvrCmj...   

                                    authorChannelUrl  \
0         http://www.youtube.com/@KaylaAzzahra-fl9sf   
1             http://www.youtube.com/@rezapratama391   
2                http://www.youtube.com/@acesabo9179   
3                  http://www.youtube.com/@WINHERO10   
4            http://www.youtube.com/@nadyaamalia1486   
...                                              ...   
28409       http://www.youtube.com/@GantplayBlogspot   
28410       http://www.youtube.com/@ramandahafiz6293   
28411                http://www.youtube.com/@e4rtz84   
28412  http://www.y

#### Menghapus data text yang duplikat

In [6]:
df = df.drop_duplicates(subset=['textOriginal'])

df.shape

(26486, 14)

#### Menghapus data author yang duplikat

In [7]:
df = df.drop_duplicates(subset=['authorChannelUrl'])

df.shape

(24317, 14)

In [8]:
df

_id                 channelId      videoId  \
0      66515de230ea1caf68a1005c  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
1      66515de230ea1caf68a1005d  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
2      66515de230ea1caf68a1005e  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
3      66515de230ea1caf68a1005f  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
4      66515de230ea1caf68a10060  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
...                         ...                       ...          ...   
28401  66515e2230ea1caf68a16f4d  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28402  66515e2230ea1caf68a16f4e  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28408  66515e2230ea1caf68a16f54  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28409  66515e2230ea1caf68a16f55  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   
28411  66515e2230ea1caf68a16f57  UC-z1ubmS-iHMVr83EEwRS2g  fNX6Gu2F7A8   

                                             textDisplay  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2      <a href="https://www.youtube.com/watch?v=fNX6G...   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28401                                              woeee   
28402                                                  1   
28408                                        =))) kerenn   
28409                                          Mantep :v   
28411                            anjing canggih waaahhhh   

                                            textOriginal  \
0                              Masih the best di hati oe   
1                                                  2024🎉   
2                                  3:34 scene yg gw suka   
3      Diantara semua YouTube rewind, tahun ini yg pa...   
4      ada yang balik kesini di 2024 ini rewind terba...   
...                                                  ...   
28401                                              woeee   
28402                                                  1   
28408                                        =))) kerenn   
28409                                          Mantep :v   
28411                            anjing canggih waaahhhh   

            authorDisplayName  \
0         @KaylaAzzahra-fl9sf   
1             @rezapratama391   
2                @acesabo9179   
3                  @WINHERO10   
4            @nadyaamalia1486   
...                       ...   
28401             @RomeuRomeu   
28402  @pricillaoktaviani2376   
28408       @is9production372   
28409       @GantplayBlogspot   
28411                @e4rtz84   

                                   authorProfileImageUrl  \
0      https://yt3.ggpht.com/ytc/AIdro_n7417ek-18wzJI...   
1      https://yt3.ggpht.com/ytc/AIdro_kt7csXE65KUFFl...   
2      https://yt3.ggpht.com/ytc/AIdro_lVQpw_BrtYzZYl...   
3      https://yt3.ggpht.com/2S9TD1tTD65kDm2AAWagbPeA...   
4      https://yt3.ggpht.com/ytc/AIdro_koGD1zvhTLlTLo...   
...                                                  ...   
28401  https://yt3.ggpht.com/ytc/AIdro_l89DpymcWykSV_...   
28402  https://yt3.ggpht.com/ytc/AIdro_lIX3-HU6zaCl0Z...   
28408  https://yt3.ggpht.com/ytc/AIdro_moVOuLXRVzZPUt...   
28409  https://yt3.ggpht.com/ytc/AIdro_nyQqIuJWFK6Ie5...   
28411  https://yt3.ggpht.com/HWmDyGnG2Kzsde0_Z8ovBbWv...   

                                    authorChannelUrl  \
0         http://www.youtube.com/@KaylaAzzahra-fl9sf   
1             http://www.youtube.com/@rezapratama391   
2                http://www.youtube.com/@acesabo9179   
3                  http://www.youtube.com/@WINHERO10   
4            http://www.youtube.com/@nadyaamalia1486   
...                                              ...   
28401             http://www.youtube.com/@RomeuRomeu   
28402  http://www.youtube.com/@pricillaoktaviani2376   
28408       http://www.youtube.com/@is9production372   
28409       http://

#### Menghapus nilai yang kosong

In [9]:
df = df.dropna()

# Melihat data yang kosong
df.isnull().sum()

_id                      0
channelId                0
videoId                  0
textDisplay              0
textOriginal             0
authorDisplayName        0
authorProfileImageUrl    0
authorChannelUrl         0
authorChannelId          0
canRate                  0
viewerRating             0
likeCount                0
publishedAt              0
updatedAt                0
dtype: int64

Dari yang awalnya berjumlah 28414 row berkurang menjadi 24317 row

#### Mengimpan Clean Comments ke MongoDB Local

In [10]:
# Koneksi ke MongoDB
client = MongoClient(local_url)
db = client[database_name]
collection = db[collection_name]

# Konversi DataFrame ke bentuk dictionary
records = df.to_dict(orient='records')

# Menyisipkan data ke MongoDB
collection.insert_many(records)

print("Data berhasil diimpor ke MongoDB Local.")

Data berhasil diimpor ke MongoDB Local.
